cli/pkg/kctrl/cmd/app/init/vendir_util.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package init
+
+import (
+	"os"
+
+	vendirconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
+	"sigs.k8s.io/yaml"
+)
+
+func ReadVendirConfig() (vendirconf.Config, error) {
+	var vendirConfig vendirconf.Config
+	exists, err := IsFileExists(VendirFileName)
+	if err != nil {
+		return vendirconf.Config{}, err
+	}
+
+	if exists {
+		vendirConfig, err = VendirConfigFromExistingFile(VendirFileName)
+		if err != nil {
+			return vendirconf.Config{}, err
+		}
+	} else {
+		vendirConfig = NewDefaultVendirConfig()
+	}
+	return vendirConfig, nil
+}
+
+func VendirConfigFromExistingFile(filePath string) (vendirconf.Config, error) {
+	content, err := os.ReadFile(filePath)
+	if err != nil {
+		return vendirconf.Config{}, err
+	}
+	vendirConfig := vendirconf.Config{}
+	err = yaml.Unmarshal(content, &vendirConfig)
+	if err != nil {
+		return vendirconf.Config{}, err
+	}
+	return vendirConfig, nil
+}
+
+func NewDefaultVendirConfig() vendirconf.Config {
+	config := vendirconf.Config{
+		APIVersion: "vendir.k14s.io/v1alpha1", // TODO: use constant from vendir package
```

Comment:
> // TODO: use constant from vendir package

the less constants the better. it's an anti-pattern to pull out strings used only in one place into constants since it introduces unnecessary indirection.

Reply:
Adopted